properties-from-ancestor-behavior.html

```diff
@@ -0,0 +1,109 @@
+```HTML
+<template>
+    myProp1 value is {{myProp1}}
+    <br/>
+    myProp2 value is {{myProp2}}
+</template>
+```
+
+while their values will come from the closest ancestor that has the corresponding dash-case attribute.
+```HTML
+<example-container-component my-prop1="some value" my-prop2="some other value"><!-- Container can also be a simple HTML <div>. As long as it has the attributes. These attributes are a requirement to discover the ancestor. If the element also has matching properties, they instead will be taken and their changes listened to. -->
+  <example-descendant-component></example-descendant-component>
+</example-container-component>
+```
+
+@polymerBehavior
+@demo demo/index.html
+*/
+    PropertiesFromAncestorBehavior = function (propsDefObject) {
+        return Object.keys(propsDefObject).map(function (propName) {
+                var propDef = propsDefObject[propName];
+
+                var ancestorWithAttribute = undefined;
+
+                var dashedName = fromCamelCaseSymbolToDash(propName);
+                var changeEventName = dashedName + '-changed';
+                var changeEventListener;
+
+                var behaviorDefinition = {
+                    properties: {},
+                    attached: function () {
+
+                        ancestorWithAttribute = polymerClosestWithAttribute(this, dashedName);
+
+                        this[propName] =
+                            !ancestorWithAttribute ?
+                                propDef.defaultValue
+                                :
+                                propName in ancestorWithAttribute ?
+                                    ancestorWithAttribute[propName]
+                                    :
+                                    ancestorWithAttribute.getAttribute(dashedName);
+
+                        if (ancestorWithAttribute) {
+                            changeEventListener = function changeEventListener() {
+                                this[propName] = ancestorWithAttribute[propName];
+                            }.bind(this);
+                            ancestorWithAttribute.addEventListener(changeEventName, changeEventListener);
+                        }
+                    },
+                    detached: function () {
+                        if (ancestorWithAttribute)
+                            ancestorWithAttribute.removeEventListener(changeEventName, changeEventListener);
+                    },
+                };
+
+
+                behaviorDefinition.properties[propName] = propDef;
+
+                return behaviorDefinition;
+            });
+
+        /// An equivalent of Element.closest('[attributeName]') for Polymer (uses `Element.domHost` to support children that are only being built, whose ancestors haven't been attached themselves yet).
+        function polymerClosestWithAttribute(e, attributeName) {
```

Comment:
Oh! Actually, sorry for misleading you by saying it can be replaced with `closest`!  
It can't because it needs to use the fallback to `Element.domHost`.